presto-main/src/main/java/io/prestosql/operator/OperatorStats.java

```diff
@@ -521,8 +524,14 @@ public OperatorStats add(Iterable<OperatorStats> operators)
             }
 
             OperatorInfo info = operator.getInfo();
-            if (base != null && info != null && base.getClass() == info.getClass()) {
-                base = mergeInfo(base, info);
+            if (base != null && info != null) {
+                if (base.getClass() != info.getClass()) {
+                    // TODO should we throw here?
+                    log.warn("Cannot merge operator info: %s and %s", base, info);
```

Comment:
i think we should actually throw here.

Reply:
Probably, that line was from 2014, so maybe original reason for this check is no longer valid

Reply:
I think we should throw.

Reply:
i replaced this with a check